src/axom/sidre/core/DataStore.cpp

```diff
@@ -618,6 +622,54 @@ bool DataStore::generateBlueprintIndex(const std::string& domain_path,
   return success;
 }
 
+#ifdef AXOM_USE_MPI
+bool DataStore::generateBlueprintIndex(MPI_Comm comm,
+                                       const std::string& domain_path,
+                                       const std::string& mesh_name,
+                                       const std::string& index_path)
+{
+  Group* domain;
+  if (domain_path == "/")
+  {
+    domain = getRoot();
+  }
+  else if (getRoot()->hasGroup(domain_path))
+  {
+    domain = getRoot()->getGroup(domain_path);
+  }
+  else
+  {
+    //There could be ranks with no domains--the MPI blueprint
+    //functions can handle this.
+    domain = nullptr;
+  }
+
+  conduit::Node mesh_node;
+  if (domain)
+  {
+    domain->createNativeLayout(mesh_node);
+  }
+
+  Group* bpindex = getRoot()->hasGroup(index_path)
+    ? getRoot()->getGroup(index_path)
+    : getRoot()->createGroup(index_path);
+
+  bool success = false;
+  conduit::Node info;
+  if(conduit::blueprint::mpi::verify("mesh", mesh_node, info, comm))
```

Comment:
Should we dump the `info` in problem cases?  Perhaps in a `SLIC_DEBUG`?

I suppose the user can do that when this function returns false...

Reply:
`info` is a local variable that will be destroyed upon return, so the user can't do anything with it.  Using SLIC is a good idea.